### PR TITLE
feat(e2e): licenses

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* Licenses mountebank mocks
* Licenses acceptance tests

## Why

Some licenses have pretty volatile dev services. By mocking all of them we eliminate potential communication issues, thus allowing us to test robustly.

* The QR code tests have to wait, bit iffy on the execution.

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced the Service Portal to display various types of licenses such as driver's, firearms, and work machine licenses, ensuring details are accessible and visible to users.
  - Expanded the types of identity documents available in the passports API to include `IdentityDocument` and `IdentityDocumentChild`.

- **Refactor**
  - Updated the `UserLicenses` component to use list elements (`<ul>` and `<li>`) for better semantic HTML structure and styling.

- **Tests**
  - Implemented end-to-end tests for the MS Licenses functionality in the Service Portal.
  - Added necessary mocks for license tests to simulate external interactions accurately.

### CODE REVIEW

File: apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
...rN1FvGiQoq+gGK6aUdHIzk7Ox5v4julkE0MhBVwQQeu09a8cjg+yTTW5OWgcxnP6H8q9L8VXStcqhOGH864XxBYSw3VtesuIrzdGc8ZdOufqpH5V5uJV1c9LCytKxRu7YSQcYP9K5y/tfmKZOzOa6hJFCFHYcevFUL6FXUjAyfevObsetY4u4sypJUYHWqhtyeMbc+3JroLy3ZMkHj07VTa23L13E9TS5ibGK8RHGfw7mgqSNwGPUVqSW2eVx9cVXaAocN36VonczsUjEFBBYjPr2qPaUHPYfw1oGEkYOAPrzUPklSRxTHYgiV2OTnI6A1Ku5mwCc+1SCI4BUBVFWo4iEBQYyO45xSuFiGCDBC8g+g5q2kZl2gfJjsO/uanhs3RcgAk9z1rUtdMKKCT8x6g0AQW9ioAaRRn1BrTgt1UDK7sDj0qWK22Mc9B6ipVVnGBwPamVYhjJYgAZPt1q2kLNgk8jn60+3tQrZbueT/AEq0EKHGAQOmBQaRXQSIELz16Zqe3j2StjGw9yOlOgiDn5sY9DU0twsShT857KKk1B4wWyzHAPAPQVXlugvyocnOAwHSmPKbjO75VB7cZpUHORjAH1rNvoaBbQLGGJY785JPcGpi53ZUAsOMgdKh2lcMSCQc8VOpwgI4GfxqLjJYLcAZXLZyTuPSr1pERjBBJI4A6VBbKcAg5Pc5rRt2WMYBwe4x3pg3Y0bUMroSVHUlsZ4rndLWbxl8R44baH7VHpMD3ByxVQ/QEkdAKm8T62PD+hzzMdp24HXkntUvwp0UWnhF9fk3Q399K7K5YqwQdjzyOOh9a7sNBVKl3sjy8XVVOLS3Z2+seLLi+tEsZi4e3RVRJTlgcZOD3U84qno6M4aRlJHy7uTlRn7w9ceneuRl1R77WE88Foicbl6jnqK6+/8AEEXhDQb3WpcRpbxl2QnPmnHy7fc16j1lboePFpanjn7VPjx7LTrLwnBcozbRPOiDG1Qcxr/M18t7yzEng+tbHjrxRd+LfEl5ql6d89zJvYjt2A/AViq/yk1MndkdLksX+sBHFaUMhC9azYSxxwfyq4CQBjpigC6JMv1qRpAmcdap+ZtAz1pJpCpUnvViZaklOPl4FRGYDODn2qu0pIxnP4VAZGkfPapYI0IJSevy+uafLOAVxjHtVKKfLbF6GmvMWVQBnJwMDjPpRLYR/9k=',
```
Comment (AndesKrrrrrrrrrrr) [inline apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts:451]: Do we need this? Can we make it a lot smaller?

File: apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts [hunk 2]
```diff
@@ -0,0 +1,581 @@
...
+  Disability,
+  DistrictCommissioners,
+  DrivingLicense,
+  EHIC,
+  Firearm,
+  HealthInsurance,
+  Hunting,
+  Passports,
+} from '../../../../../../../../infra/src/dsl/xroad'
+import { DriverLicenseDto } from '@island.is/clients/driving-license'
```
[the remark this answers is not in the document]
Comment (thorkellmani) [reply apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts:14]: ah yeah, sorry

File: apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts (hunk 1 above)
Comment (thorkellmani) [reply apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts:451]: probably unnecessary, i'll remove

File: apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (thorkellmani) [reply apps/system-e2e/src/tests/islandis/service-portal/acceptance/mocks/licenses.mock.ts:14]: mostly used for type checking while creating the mocks

**Review by Valur — APPROVED** [review]

**Pull request merged**